src/server/system_services/pool_server.js

```diff
@@ -88,12 +88,26 @@ function set_pool_controller_factory(pool_controller_factory) {
 
 }
 
-function new_pool_defaults(name, system_id, resource_type, pool_node_type) {
+// The function checks whether the pool/resource has an owner
+// and only allows deletion in case that the owner is also the requester of the deletion
+function check_deletion_ownership(req, resource_owner_id) {
+    if (config.RESTRICT_RESOURCE_DELETION) {
+        const requester_is_sys_owner = String(req.account._id) === String(req.system.owner._id);
+        if (!requester_is_sys_owner && String(resource_owner_id) !== String(req.account._id)) {
+            throw new RpcError('UNAUTHORIZED', 'The pool or resource can be deleted only by its owner or the system administrator');
```

Comment:
Print an error message to log including info on the req account vs the owner account